### PR TITLE
[Quill] update to v1.3.1

### PR DESCRIPTION
**Pull request by odygrd**
**Describe the pull request**

- What does your PR fix? Fixes 
Version update

- Which triplets are supported/not supported? Have you updated the CI baseline?
No change in the CI baseline

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -1541,7 +1541,6 @@ quickfix:x64-windows-static=ignore
 quickfix:x86-windows=ignore
 quickfix:x64-uwp=fail
 quickfix:arm-uwp=fail
-quill:arm64-windows=fail
```
Comment (LilyWangL) [inline scripts/ci.baseline.txt:1544]: This command shouldn't be removed, because this port doesn't support `arm`.

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: scripts/ci.baseline.txt (hunk 1 above)
Comment (odygrd) [reply scripts/ci.baseline.txt:1544]: Hey Lily, the 1.3.1 version now supports arm64, should I still add this line back ?

**Comment by LilyWangL** [comment]
If the new version supports arm64, you need remove `arm` from  `Supports` in CONTROL and remove `ON_ARCH "arm"` from `vcpkg_fail_port_install` in portfile.cmake. Because `arm - VCPKG_TARGET_ARCHITECTURE == "arm" or VCPKG_TARGET_ARCHITECTURE == "arm64"`.
And I test this PR on my locally, I build `quill:arm64-windows` failed with this error:
```
vcpkg\buildtrees\quill\src\76aaafff6d-cac280987f\quill\include\quill/detail/misc/Rdtsc.h(62): error C3861: '__rdtsc': identifier not found
vcpkg\buildtrees\quill\src\76aaafff6d-cac280987f\quill\include\quill/detail/misc/Rdtsc.h(62): error C3861: '__rdtsc': identifier not found
vcpkg\buildtrees\quill\src\76aaafff6d-cac280987f\quill\include\quill/detail/misc/Rdtsc.h(62): error C3861: '__rdtsc': identifier not found
vcpkg\buildtrees\quill\src\76aaafff6d-cac280987f\quill\include\quill/detail/misc/Rdtsc.h(62): error C3861: '__rdtsc': identifier not found
vcpkg\buildtrees\quill\src\76aaafff6d-cac280987f\quill\include\quill/detail/misc/Rdtsc.h(62): error C3861: '__rdtsc': identifier not found
```

**Comment by odygrd** [comment]
thanks Lilly, I am reverting the changes for now

**Review by LilyWangL — APPROVED** [review]

**Comment by strega-nil** [comment]
Alright cool, thanks y'all!

**Pull request merged**